ivy/functional/frontends/numpy/statistics/order_statistics.py

```diff
@@ -39,6 +43,82 @@ def _cpercentile(N, percent, key=lambda x: x):
     return d0 + d1
 
 
+
+@to_ivy_arrays_and_back
+@handle_numpy_out
+def percentile(a, q, axis=None, out=None, overwrite_input=False, interpolation='linear', keepdims=False):
+    a = ivy.array(a)
+    q = ivy.divide(q, 100.0)
+    q = ivy.array(q)
+
+    if not _quantile_is_valid(q):
+        ivy.logging.warning("percentile s must be in the range [0, 100]")
+        return []
+
+    if axis is None:
+        resultarray = []
+        nanlessarray = ivy.where(~ivy.isnan(a), a, ivy.array(float('nan')))
+        for i in q:
+            resultarray.append(_cpercentile(nanlessarray, i, interpolation))
+        return resultarray
+    elif axis == 1:
+        resultarray = []
+        nanlessarrayofarrays = ivy.stack([ivy.where(~ivy.isnan(row), row, ivy.array(float('nan'))) for row in a])
+        for i in q:
+            arrayofpercentiles = []
+            for ii in nanlessarrayofarrays:
+                arrayofpercentiles.append(_cpercentile(ii, i, interpolation))
+            resultarray.append(arrayofpercentiles)
+        return resultarray
+    elif axis == 0:
+        try:
+            a = ivy.swapaxes(a, 0, 1)
+        except ivy.exceptions.IvyError:
+            ivy.logging.warning("axis is 0 but couldn't swap")
+
+        finally:
+            resultarray = []
+            nanlessarrayofarrays = ivy.stack([ivy.where(~ivy.isnan(col), col, ivy.array(float('nan'))) for col in a])
+            for i in q:
+                arrayofpercentiles = []
+                for ii in nanlessarrayofarrays:
+                    arrayofpercentiles.append(_cpercentile(ii, i, interpolation))
+                resultarray.append(arrayofpercentiles)
+            return resultarray
+
+def _quantile_is_valid(q):
+    return ivy.all(ivy.logical_and(q >= 0, q <= 1))
+
+def _cpercentile(arr, q, interpolation='linear'):

```

Comment:
It seems `_cpercentile` function already exists: https://github.com/unifyai/ivy/blob/main/ivy/functional/frontends/numpy/statistics/order_statistics.py#L13